compiler/rustc_passes/src/check_attr.rs

```diff
@@ -2005,12 +2001,36 @@ impl CheckAttrVisitor<'_> {
             }
         };
 
-        match (meta_item.name_or_empty(), meta_item.value_str()) {
-            (sym::natvis_file, Some(_)) => true,
+        let visualizer_path = match (meta_item.name_or_empty(), meta_item.value_str()) {
+            (sym::natvis_file, Some(value)) => value,
+            (sym::gdb_script_file, Some(value)) => value,
             (_, _) => {
                 self.emit_debugger_visualizer_err(attr);
-                false
+                return false;
             }
+        };
+
+        let file =
+            match resolve_path(&self.tcx.sess.parse_sess, visualizer_path.as_str(), attr.span) {
+                Ok(file) => file,
+                Err(mut err) => {
+                    err.emit();
+                    return false;
+                }
+            };
+
+        if file.is_file() && std::fs::read(&file).is_ok() {
+            true
+        } else {
```

Comment:
I removed the `is_file()` check since it's redundant but I'd like to read the file anyway. Thus, if the file cannot be read, this can be caught at an early stage as opposed to waiting for the query to be used which might not happen for certain crates even if the attribute is used.